pkg/chunk/aws/dynamodb_table_client.go

```diff
@@ -290,23 +312,35 @@ func (d dynamoTableClient) DescribeTable(ctx context.Context, name string) (desc
 }
 
 func (d dynamoTableClient) UpdateTable(ctx context.Context, current, expected chunk.TableDesc) error {
-	var err error
-	if !current.WriteScale.Enabled {
-		if expected.WriteScale.Enabled {
-			err = d.enableAutoScaling(ctx, expected)
-		}
-	} else {
-		if !expected.WriteScale.Enabled {
-			err = d.disableAutoScaling(ctx, expected)
-		} else if current.WriteScale != expected.WriteScale {
-			err = d.enableAutoScaling(ctx, expected)
+	if expected.WriteScale.Enabled && d.metrics != nil {
```

Comment:
Perhaps the check `expected.WriteScale.Enabled && d.metrics != nil` should be put in the constructor so we can fail early? Then we can just check `expected.WriteScale.Enabled` here.

Reply:
Actually, in what situation is metrics nil?

Reply:
metrics is nil when you want AWS auto-scaling

Reply:
I don't see how - must be missing something.  Would you mind explaining a little further for me?

Reply:
plus or minus a bug...

Reply:
I fixed the bug.